### PR TITLE
10 criteria form

### PR DESCRIPTION
**Pull request by RainbowGiantSquid**
navigate to criteria/new 
submit form (look at rails server - you should see writing to db)
redirect

### CODE REVIEW

File: app/javascript/components/progress/MockProgressBar.vue [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+<script>
+export default {
+  name: "MockProgressBar",
+  props: {
+    title: {
+      type: Object,
+      required: true,
+    },
+  },
```
Comment (RainbowGiantSquid) [inline app/javascript/components/progress/MockProgressBar.vue:18]: Sorry - this component is a bit basic and could've been probably done better.

File: app/javascript/packs/application.js [hunk 2]
```diff
@@ -37,6 +48,8 @@ document.addEventListener('DOMContentLoaded', () => {
       store,
       components: {
 	      FilteredTable,
```
Comment (sergiomarrocoli) [inline app/javascript/packs/application.js:50]: whitespace

File: app/models/services/criterium_props.rb [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+              {
+                type: 'radiogroup',
+                name: 'boundary',
+                title: I18n.t('form.criteria.q1.label'),
+                isRequired: true,
+                choices: [
+                  {
+                    value: 'true',
+                    text:
+                      'The area has, or will have, a clearly defined boundary'
```
Comment (sergiomarrocoli) [inline app/models/services/criterium_props.rb:33]: think these should be in the locales

File: config/locales/form/en.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      q2:
+        label: Please select the objectives of the Convention on Biological Diversity that your commitment contributes to
+      q3:
+        label: Who is making the commitment?
+      q4:
+        label: Is the duration of the commitment at least 5 years?
+      q5:
+        label: Are you able to submit annual progress reports to this platform?
+    commitments: 
+
```
Comment (sergiomarrocoli) [inline config/locales/form/en.yml:21]: whitespace

File: app/models/services/criterium_props.rb [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+              },
+              {
+                type: 'radiogroup',
+                name: 'progress_report',
+                title: I18n.t('form.criteria.q5.label'),
+                isRequired: true,
+                choices: [
+                  {
+                    value: 'true',
+                    text: 'Yes'
```
Comment (sergiomarrocoli) [inline app/models/services/criterium_props.rb:95]: locales too i guess, just 'true' and 'false' keys under form, rather than question specific

**Comment by sergiomarrocoli** [comment]
looks good to me! just some formatting stuff, and some things that could be moved to locales

File: app/javascript/components/forms/SurveyForm.vue [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+  mounted() {
+    setAxiosHeaders(axios);
+  },
+
+  methods: {
+    onComplete(sender) {
+      const options = {
+        method: this.formData.config.method,
+        data: { [this.formData.config.root_key]:  sender.data },
```
Comment (yndajas) [inline app/javascript/components/forms/SurveyForm.vue:48]: Double whitespace between key and value here

File: app/javascript/components/forms/SurveyForm.vue [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        .then((response) => {
+          this.redirect(response.data.redirect_path)
+        })
+        .catch((error) => {
+          console.log("FAILED!", error.data);
+        })
+    },
+    redirect(link) {
+      if (link) {
+        Turbolinks.visit(`${link}`);
```
Comment (yndajas) [inline app/javascript/components/forms/SurveyForm.vue:60]: I think the argument here can just be `link` without the interpolation?

**Review by yndajas — CHANGES_REQUESTED** [review]
Looks really good! Mostly just a few formatting and naming suggestions

File: app/assets/stylesheets/components/_progress.scss [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+      border-style: solid;
+      border-width: 41px 0 41px 55px;
+      border-color: transparent transparent transparent $green-dark;
+    }
+  }
+
+  &__dormant {
+    margin-left: 4rem;
+    padding: 2rem;
+    &:before {
```
[the remark this answers is not in the document]
Comment (yndajas) [reply app/assets/stylesheets/components/_progress.scss:48]: As in whitespace before the nested rule

File: app/assets/stylesheets/components/_progress.scss [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+//--------------------------------------------------
+
+//--------------------------------------------------
+// mixins
+//--------------------------------------------------
+
+//--------------------------------------------------
+// classes
+//--------------------------------------------------
+.progress {
```
[the remark this answers is not in the document]
Comment (yndajas) [reply app/assets/stylesheets/components/_progress.scss:12]: The grey background seems to have gone, so maybe it was a glitch 🤔

File: app/javascript/components/forms/SurveyForm.vue [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+import Turbolinks from "turbolinks";
+import axios from "axios";
+import { setAxiosHeaders } from "../../helpers/axios-helpers";
+import "survey-vue/modern.css";
+SurveyVue.StylesManager.applyTheme("modern");
+
+var Survey = SurveyVue.Survey;
+
+export default {
+  name: "SurveyForm",
```
[the remark this answers is not in the document]
Comment (RainbowGiantSquid) [reply app/javascript/components/forms/SurveyForm.vue:18]: The plan is for both forms to use the same component. I'll explain in stand-up

File: config/locales/form/en.yml [hunk 11]
```diff
@@ -0,0 +1,27 @@
+en:
+  form:
+    criteria:
+      progress: 
+         title: 
+            active: "Criteria"
```
[the remark this answers is not in the document]
Comment (RainbowGiantSquid) [reply config/locales/form/en.yml:6]: yep, sorry - copy-pasting slackness on my side

File: config/locales/form/en.yml [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+            dormant: "Commitment form"
+      title: 'Form 1: criteria'
+      description: 'Form 1 will help establish whether this platform is the right place for your commitment. ‘Commitment’ refers to what you intend to do, for example, commitments to enhance/improve an existing area where new actions are planned or to establish a new area. ‘Area’ refers to the geographical area this is (or will be) under management as a result of this commitment.'
+      other: None of the above
```
[the remark this answers is not in the document]
Comment (RainbowGiantSquid) [reply config/locales/form/en.yml:10]: changed

File: app/javascript/components/progress/MockProgressBar.vue [hunk 13]
```diff
@@ -0,0 +1,22 @@
+<template>
+  <div class="container">
+    <ul class="progress">
+      <li class="progress__active">{{ title.active }}</li>
+      <!-- the span has a peudoelement attached for styling -->
+      <li class="progress__dormant"><span></span>{{ title.dormant }}</li>
```
[the remark this answers is not in the document]
Comment (RainbowGiantSquid) [reply app/javascript/components/progress/MockProgressBar.vue:6]: yep, as in comment above (fixed typo) it has a pseudoelement for before border

[the remark this answers is not in the document]
Comment (yndajas) [reply app/javascript/components/progress/MockProgressBar.vue:6]: Ah, that's probably why I saw that grey background (I tried deleting it)!

File: package.json [hunk 14]
```diff
@@ -4,12 +4,22 @@
   "dependencies": {
```
[the remark this answers is not in the document]
Comment (RainbowGiantSquid) [reply package.json:4]: I removed vue-router, jquery unfortunately has to stay for the correct display of inputs.

File: app/models/services/criterium_props.rb [hunk 15]
```diff
@@ -0,0 +1,108 @@
+class Services::CriteriumProps
+  def initialize(criterium)
+    @criterium = criterium
+  end
+
+  def call
+    {
+      config:
+      {
+        action: '/criteria.json',
```
[the remark this answers is not in the document]
Comment (RainbowGiantSquid) [reply app/models/services/criterium_props.rb:10]: Needs to have .json atm or it returns html format in response

File: app/javascript/components/progress/MockProgressBar.vue (hunk 13 above)
[the remark this answers is not in the document]
Comment (RainbowGiantSquid) [reply app/javascript/components/progress/MockProgressBar.vue:6]: Yeah, I know, it's not a really elegant way of doing this. I'm open to suggestions.

File: app/models/services/criterium_props.rb (hunk 15 above)
[the remark this answers is not in the document]
Comment (sergiomarrocoli) [reply app/models/services/criterium_props.rb:10]: yeah, with rails depending on what the request type is, we can return different responses from a single controller action with this method in the controllers: 
```
    respond_to do |format|
      format.html { whatever you want this to do, usually used for standard page reload with a new view }
      format.json { some json for use by a component }
      format.xml { render :xml => @people.to_xml }
    end
```
the default is render the corresponding view, so commitment#new action will render views/commitments/new.html.erb

https://apidock.com/rails/ActionController/MimeResponds/InstanceMethods/respond_to

File: app/assets/stylesheets/components/_progress.scss [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+  @include flex;
+  @include flex-v-center;
+  width: 100%;
```
Comment (yndajas) [inline app/assets/stylesheets/components/_progress.scss:15]: I think this one goes at the end

File: app/assets/stylesheets/components/_progress.scss [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+      margin-top: -2rem;
+      margin-left: 2rem;
+      z-index: 1;
+
+      content: '';
+      position: absolute;
+
+      border-style: solid;
+      border-width: 2.563rem 0 2.563rem 3.438rem;
+      border-color: transparent transparent transparent $green-dark;
```
Comment (yndajas) [inline app/assets/stylesheets/components/_progress.scss:38]: I think the order/spacing here should be:

```
border-color: transparent transparent transparent $green-dark;
border-style: solid;
border-width: 2.563rem 0 2.563rem 3.438rem;
content: '';
margin-top: -2rem;
margin-left: 2rem;
width: 0; height: 0;

position: absolute;
z-index: 1;
```

File: app/assets/stylesheets/components/_progress.scss [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+      width: 0; height: 0;
+
+      border-color: transparent transparent transparent $white;
+      border-style: solid;
+      border-width: 2.563rem 0 2.563rem 3.438rem;
+      content: '';
+      margin-top: -2rem;
+      margin-left: -5rem;
+      position: absolute;
+      z-index: 0;
```
Comment (yndajas) [inline app/assets/stylesheets/components/_progress.scss:56]: As above re: order/spacing

File: app/assets/stylesheets/components/_progress.scss [hunk 19]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    span:before {
+      width: 1rem; height: 6rem;
```
Comment (yndajas) [inline app/assets/stylesheets/components/_progress.scss:60]: Think this goes just before the gap (after margin)

File: app/javascript/components/progress/MockProgressBar.vue (hunk 13 above)
[the remark this answers is not in the document]
Comment (yndajas) [reply app/javascript/components/progress/MockProgressBar.vue:6]: I have no idea how triangles work in CSS!

File: app/models/services/criterium_props.rb (hunk 15 above)
[the remark this answers is not in the document]
Comment (yndajas) [reply app/models/services/criterium_props.rb:10]: Oh nice!

**Review by yndajas — CHANGES_REQUESTED** [review]
Just a few style ordering things but otherwise I think it's good to go

File: app/assets/stylesheets/components/_progress.scss (hunk 17 above)
Comment (RainbowGiantSquid) [reply app/assets/stylesheets/components/_progress.scss:38]: oh ok, sorry, worked mainly with tailwind so far 😅

**Review by yndajas — APPROVED** [review]

**Pull request merged**